pubsub/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/GooglePubSubSink.scala

```diff
@@ -98,11 +111,28 @@ class GooglePubSubSink private (val maxBytes: Int, publisher: Publisher, project
   private def eventToPubsubMessage(event: Array[Byte]): PubsubMessage =
     PubsubMessage.newBuilder.setData(ByteString.copyFrom(event)).build()
 
+  private def retryRunnable: Runnable = new Runnable {
+    override def run() {
+      val failedInserts = failedInsertsBuffer.synchronized {
+        val records = failedInsertsBuffer.toList
+        failedInsertsBuffer.clear()
+        records
+      }
+      if (failedInserts.nonEmpty) {
+        log.info(s"Retrying to insert ${failedInserts.size} records into $topicName")
+        storeRawEvents(failedInserts, "NOT USED")
+      }
+      scheduledExecutor.schedule(retryRunnable, retryInterval.toMillis, MILLISECONDS)
```

Comment:
Instead of this line, did you consider using `scheduledExecutor.scheduleWithFixedDelay`?